faw/ci/ci_container_service.py

```diff
@@ -808,6 +875,13 @@ def _create_dockerfile_contents(development, config, config_data, build_dir, bui
             COPY {shlex.quote(config_rel_dir)} /home/dist
             ''')
 
+    # Dependencies between devmounts and stages are
+    # inferred from commands, clear out the current state
+    # as we will be collecting this again.
+    # TODO: This need some more thought.
```

Comment:
We derive the relationship between devmounts and stages by looking at the commands in each stage. (We do this so that we know which stages to rebuild when a devmount updates). A potential concern here is that the stages and stage commands are subject to change during runtime; for example, if someone uses the CI features to push a new configuration file to the FAW. This means that a CI command can not only introduce new devmount dependencies, but also remove them. 

Note that nearly everything else wrt devmounts - introduction of devmounts, parser - devmount dependencies etc are *static* and read and acted upon only a single time when the CI container starts.

This interaction between CI and devmount features may need more thought in order to streamline things in the future, depending on how this feature gets used.